compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2193,14 +2193,8 @@ private STNode parseTypeDescriptor(ParserRuleContext context, boolean isTypedBin
     private STNode parseTypeDescriptor(List<STNode> qualifiers, ParserRuleContext context,
                                        boolean isTypedBindingPattern, boolean isInConditionalExpr) {
         startContext(context);
-        STNode typeDesc = parseTypeDescriptorInternal(qualifiers, context, isTypedBindingPattern, isInConditionalExpr);
-        endContext();
-        return typeDesc;
-    }
-
-    private STNode parseTypeDescriptorInternal(List<STNode> qualifiers, ParserRuleContext context,
-                                               boolean isTypedBindingPattern, boolean isInConditionalExpr) {
         STNode typeDesc = parseTypeDescriptorInternal(qualifiers, context, isInConditionalExpr);
+        endContext();
```

Comment:
We will lose the original context if we end the context before we complete parsing the complete type descriptor. 
eg `type myint int[5;` error handler will look for expression after recovering `]` where it should look for the semicolon.